src/main/java/io/cdap/plugin/http/source/batch/HttpBatchSourceConfig.java

```diff
@@ -29,6 +50,58 @@ protected HttpBatchSourceConfig(String referenceName) {
   @Override
   public void validate(FailureCollector failureCollector) {
     super.validate(failureCollector);
+    validateCredentials(failureCollector);
+  }
+
+  public void validateCredentials(FailureCollector collector) {
+    try {
+      if (getAuthType() == AuthType.OAUTH2) {
+        validateOAuth2Credentials(collector);
+      } else if (getAuthType() == AuthType.BASIC_AUTH) {
+        validateBasicAuthCredentials(collector);
+      }
+    } catch (IOException e) {
+      throw new IllegalStateException("Unable to authenticate the given info", e);
+    }
+  }
+
+  private void validateOAuth2Credentials(FailureCollector collector) throws IOException {
+    try {
+      HttpClientBuilder httpclientBuilder = HttpClients.custom();
+      if (!containsMacro(PROPERTY_PROXY_URL) && !Strings.isNullOrEmpty(getProxyUrl())) {
+        HttpHost proxyHost = HttpHost.create(getProxyUrl());
+        if (!Strings.isNullOrEmpty(getProxyUsername()) && !Strings.isNullOrEmpty(getProxyPassword())
+          && !containsMacro(PROPERTY_PROXY_USERNAME) && !containsMacro(PROPERTY_PROXY_PASSWORD)) {
+          CredentialsProvider credsProvider = new BasicCredentialsProvider();
+          credsProvider.setCredentials(new AuthScope(proxyHost),
+            new UsernamePasswordCredentials(getProxyUsername(), getProxyPassword()));
+          httpclientBuilder.setDefaultCredentialsProvider(credsProvider);
+        }
+        httpclientBuilder.setProxy(proxyHost);
+      }
+      CloseableHttpClient closeableHttpClient = httpclientBuilder.build();
+      AccessToken accessToken = OAuthUtil.getAccessTokenByRefreshToken(closeableHttpClient, this);
+    } catch (JsonSyntaxException | HttpHostConnectException e) {
+      throw new IllegalStateException("Error executing this request: " + e.getMessage(), e);
```

Comment:
We can wrap the exception message and display the error on the UI using the failure collector